### PR TITLE
ED-2003 better debug on failure

### PR DESCRIPTION
**Pull request by kowalcj0**
This [ticket](https://uktrade.atlassian.net/browse/ED-2003)

This will print out information about last:
* request URL, headers & body
* redirects (if present)
* response headers & content

This should help to quickly understand why any given scenario failed.

### CODE REVIEW

**Comment by sdonk** [comment]
`termcolor` for the output?

https://pypi.python.org/pypi/termcolor

**Comment by kowalcj0** [comment]
@sdonk after playing with `termcolor` for a while, I've decide to use it.

File: tests/functional/features/environment.py [hunk 1]
```diff
@@ -10,22 +10,56 @@
...
-from tests.functional.features.utils import init_loggers
+from tests.functional.features.utils import blue, green, init_loggers, red
 
 
 def before_step(context, step):
     logging.debug('Step: %s %s', step.step_type, str(repr(step.name)))
 
 
 def after_step(context, step):
+    from pprint import pformat
```
Comment (richtier) [inline tests/functional/features/environment.py]: this should be imported at module level

Comment (kowalcj0) [inline tests/functional/features/environment.py]: ah yes, spot on. thx

**Pull request merged**